src/Raven.Server/Documents/Indexes/MapReduce/Static/MapReduceIndex.cs

```diff
@@ -30,6 +30,8 @@ public class MapReduceIndex : MapReduceIndexBase<MapReduceIndexDefinition>
 
         private readonly Dictionary<string, AnonymousObjectToBlittableMapResultsEnumerableWrapper> _enumerationWrappers = new Dictionary<string, AnonymousObjectToBlittableMapResultsEnumerableWrapper>();
 
+        public PropertyAccessor OutputReduceToCollectionPropertyAccessor;
```

Comment:
move it to LuceneIndexPersistance, cache it there, and pass as a parameter to OutputReduceIndexWriteOperation?